app/models/manageiq/providers/openshift/infra_manager.rb

```diff
@@ -0,0 +1,74 @@
+ManageIQ::Providers::Kubevirt::InfraManager.include(ActsAsStiLeafClass)
+
+class ManageIQ::Providers::OpenShift::InfraManager < ManageIQ::Providers::Kubevirt::InfraManager
+  belongs_to :parent_manager,
+             :foreign_key => :parent_ems_id,
+             :class_name  => "ManageIQ::Providers::ContainerManager"
```

Comment:
This should be `ManageIQ::Providers::Openshift::ContainerManager`, then you can add an invere_of :infra_manager